File: e13/java/Solution.java
class Solution {
    /**
     * Returns a index to the first occurrence of target in source,
     * or -1  if target is not part of source.
     * @param source string to be scanned.
     * @param target string containing the sequence of characters to match.
     */
    public static int strStr(String source, String target) {
        //write your code here
		if (source == null || target == null)
			return -1;
		int lenSrc=source.length();
		int lenTgt=target.length();

		if (lenTgt == 0)
			return 0;
		if (lenSrc < lenTgt)
			return -1;

		int equalLen = 0;
		int index = 0;
		for(int i = 0; i <= lenSrc - lenTgt; i++) {
			index = i;
			equalLen = 0;
			for (int j=0; j < lenTgt; j++) {
				if (source.charAt(i+j) == target.charAt(j)) {
					equalLen++;
				}
			}
			if (equalLen == lenTgt) {
				break;
			}
		}
		if (equalLen == lenTgt)
			return index;
		else
			return -1;
    }

	public static void main(String[] args) {
		int result = strStr(args[0], args[1]);
		System.out.println(result);
		System.out.println(strStr(null, "a"));
	}
}

File: e13/python/solution.py
import sys


class Solution:
    def strStr(self, source, target):
        # write your code here
        if source is None or target is None:
            return -1

        if target == "":
            return 0
        if len(source) < len(target):
            return -1
        index = 0
        equal = 0
        for i in range(len(source) - len(target) + 1):
            index = i
            equal = 0
            for j in range(len(target)):
                if source[i+j] == target[j]:
                    equal += 1
                else:
                    break
            if equal == len(target):
                return index
        return -1
#       return source.find(target)

def main(argv=None):
    if argv is None:
        argv = sys.argv
    print Solution().strStr("abcdefg", "bcd")
    print Solution().strStr("abcdefg", "fg")
    print Solution().strStr("abcdefg", "g")
    print Solution().strStr("", "bcd")
    print Solution().strStr("abcd", "")
    print len(sys.argv)

if __name__ == "__main__":
    sys.exit(main())
